ts/features/pn/aar/saga/fetchNotificationDataSaga.ts

```diff
@@ -0,0 +1,156 @@
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { StackActions } from "@react-navigation/native";
+import { sequenceS } from "fp-ts/lib/Apply";
+import * as E from "fp-ts/lib/Either";
+import * as O from "fp-ts/lib/Option";
+import { pipe } from "fp-ts/lib/function";
+import { call, put, select } from "typed-redux-saga/macro";
+import { MessageSubject } from "../../../../../definitions/backend/MessageSubject";
+import { ThirdPartyMessage as PnThirdPartyMessage } from "../../../../../definitions/pn/ThirdPartyMessage";
+import { ThirdPartyMessage as AarThirdPartyMessage } from "../../../../../definitions/pn/aar/ThirdPartyMessage";
+import NavigationService from "../../../../navigation/NavigationService";
+import { pnMessagingServiceIdSelector } from "../../../../store/reducers/backendStatus/remoteConfig";
+import { SessionToken } from "../../../../types/SessionToken";
+import { isTestEnv } from "../../../../utils/environment";
+import { MESSAGES_ROUTES } from "../../../messages/navigation/routes";
+import { getServiceDetails } from "../../../messages/saga/utils";
+import { profileFiscalCodeSelector } from "../../../settings/common/store/selectors";
+import PN_ROUTES from "../../navigation/routes";
+import { SendAARClient } from "../api/client";
+import {
+  EphemeralAarMessageDataActionPayload,
+  populateStoresWithEphemeralAarMessageData,
+  setAarFlowState
+} from "../store/actions";
+import { currentAARFlowData } from "../store/reducers";
+import { fillerEphemeralAARMarkdown } from "../utils/detailsById";
+import { AarFlowStates, sendAARFlowStates } from "../utils/stateUtils";
+
+export function* fetchAarDataSaga(
+  fetchData: SendAARClient["getAARNotification"],
+  sessionToken: SessionToken
+) {
+  const currentState = yield* select(currentAARFlowData);
+  if (currentState.type !== sendAARFlowStates.fetchingNotificationData) {
+    return;
+  }
+  try {
+    const result = yield* call(fetchData, {
+      Bearer: sessionToken,
+      iun: currentState.iun,
+      mandateId: currentState.mandateId
+    });
+
+    if (E.isLeft(result)) {
+      yield* put(
+        setAarFlowState({
+          type: sendAARFlowStates.ko,
+          previousState: currentState
+        })
+      );
+      return;
+    }
+    const { status, value } = result.right;
+    if (status === 200) {
+      const payload = yield* call(
+        aarMessageDataPayloadFromResponse,
+        value,
+        currentState
+      );
+
+      if (O.isNone(payload)) {
+        return;
+      }
+      const data = payload.value;
+
+      yield* put(populateStoresWithEphemeralAarMessageData(data));
+
+      yield* call(
```

Comment:
I think that this navigation does not belong here. Mixing navigation between screens and sagas leads to having an hybrid behaviour that later explodes into a lot of places where different cases are handled. The screen that is displaying the loading spinner should be responsible for detecting the change in the state and trigger the navigation